### PR TITLE
Don't open a new page when resending a verification email.

### LINKED ISSUE #585: Resend verification email link should be an action, rather than an about:blank link
It being `about:blank` means it gets opened in the user's browser if they click it in the electron app, which may or may not be logged in. Pretty non-ideal experience.

Instead we should handle the click and `fetch` the right URL, probably show the result in a snackbar.

### PR DESCRIPTION
**Pull request by 2Pacalypse-**
Instead, show the success/failure of the action in a snackbar.

Fixes #585

### CODE REVIEW

File: client/auth/action-creators.ts [hunk 1]
```diff
@@ -17,7 +18,7 @@ function idRequest<
   ActionType extends Extract<IdRequestable, { type: ActionTypeName }>
 >(
   type: ActionTypeName,
-  fetcher: () => Promise<ActionType['payload']>,
+  fetcher: (dispatch: any) => Promise<ActionType['payload']>,
```
Comment (2Pacalypse-) [inline client/auth/action-creators.ts:21]: Could use some help with typing the `dispatch` function here.

Comment (tec27) [reply client/auth/action-creators.ts:21]: sendVerificationEmail isn't an ID'd request (it's not associated with a form, we'd never show the error from the reducer, so why do this?), so just don't add this and don't use this function for it and you won't have this problem.

For future reference though, `dispatch-registry` exports a typing for the dispatch function.

File: client/auth/action-creators.ts [hunk 2]
```diff
@@ -135,7 +136,26 @@ export function verifyEmail(token: string) {
 export function sendVerificationEmail() {
   const url = '/api/1/users/sendVerification'
 
-  return idRequest('@auth/sendVerificationEmail', () => fetch<void>(url, { method: 'post' }))
+  return idRequest('@auth/sendVerificationEmail', dispatch =>
+    fetch<void>(url, { method: 'post' }).then(
+      () =>
+        dispatch(
+          openSnackbar({
+            message: 'Verification email has successfully been sent. Check your email.',
```
Comment (tec27) [inline client/auth/action-creators.ts:144]: Remove the second sentence here

File: client/auth/action-creators.ts [hunk 3]
```diff
@@ -135,7 +136,26 @@ export function verifyEmail(token: string) {
...
+            time: TIMING_LONG,
+          }),
+        ),
+      err => {
+        dispatch(
+          openSnackbar({
+            message: 'Something went wrong while trying to send a verification email.',
```
Comment (tec27) [inline client/auth/action-creators.ts:151]: Replace with `Something went wrong while sending a verification email, please try again later.`

File: client/auth/action-creators.ts (hunk 2 above)
Comment (tec27) [reply client/auth/action-creators.ts:144]: Probably also `Verification email has been sent successfully.`

File: client/auth/action-creators.ts [hunk 4]
```diff
@@ -132,10 +133,27 @@ export function verifyEmail(token: string) {
...
+        ),
+      err => {
+        dispatch(
+          openSnackbar({
+            message:
+              'Something went wrong while sending a verification email, please try again later.',
+            time: TIMING_LONG,
+          }),
+        )
+        throw err
```
Comment (tec27) [inline client/auth/action-creators.ts:154]: Probably don't wanna throw this, it'll count as an uncaught error (and you *did* handle it here)

**Review by tec27 — APPROVED** [review]

**Pull request merged**